### ISSUE TITLE
Are Array Prototype Methods Supported?

### ISSUE THREAD

**Issue by matthewwithanm** [amilajack/eslint-plugin-compat#540]
Hey! I'm very confused about the status of Array prototype methods. There are a bunch of open issues about them with conflicting information about how/whether these are meant to be supported. The README specifically calls out using `Array.prototype.push()`, but I can't seem to get errors for `.flatMap()`, or `.flat()`.

I was going to comment on the recently opened #539 about `.at()`, but after some troubleshooting it seems like maybe that's a separate issue? For `x.flat()`, the check is reaching [this code][1] but not matching because `objectName` is `"x"`:

```ts
const failingRule = rules.find(
  (rule) =>
    rule.object === objectName &&
    (rule.property == null || rule.property === propertyName)
);
```

For `[].flat()`, it hits [this branch][2] but doesn't match because `protoChainId` is `".flat"`, instead of the rule's `"Array.flat"`:

```ts
const protoChain =
  firstObj === "window" || firstObj === "globalThis"
    ? rawProtoChain.slice(1)
    : rawProtoChain;
const protoChainId = protoChain.join(".");
const failingRule = rules.find(
  (rule) => rule.protoChainId === protoChainId
);
```

`.at()` (#539) *seems* to have a different problem in that there's no rule for it in the list.

What's the expected behavior for these two cases? Should they cause failures? Is this a bug, or is the README just out of date?

Thanks for the help!

[1]: https://github.com/amilajack/eslint-plugin-compat/blob/ba0a4e9d1229f6ed52100d725272cd1429923bda/src/helpers.ts#L141-L145
[2]: https://github.com/amilajack/eslint-plugin-compat/blob/ba0a4e9d1229f6ed52100d725272cd1429923bda/src/helpers.ts#L122-L129

**Comment by koddsson** [comment]
Is this still not working in the latest version (4.1.4)? I think I've fixed it there. If not still working, can you provide a small reproduction case that we can take a look at?

**Comment by chenyulun** [comment]
同意，不能写与ecmascript内置方法名称作为自定义名称，这样就可以简单的使用`.flat`来检测,

其实前面的变量可以是任意变量所以丢失了跟踪，也能是函数的入参
